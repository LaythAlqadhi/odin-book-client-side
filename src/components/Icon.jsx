import React from 'react';
import PropTypes from 'prop-types';

function Icon({ className, size, path }) {
  const [width, height] = [size];

  return (
    <svg
      className={className}
      aria-hidden="true"
      width={width}
      height={height}
      viewBox="0 0 24 24"
    >
      <path
        className={`${className} transition-transform ease-in-out active:scale-95`}
        d={path}
      />
    </svg>
  );
}

Icon.defaultProps = {
  className: '',
  size: '32',
};

Icon.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string,
  path: PropTypes.string.isRequired,
};

export default Icon;
