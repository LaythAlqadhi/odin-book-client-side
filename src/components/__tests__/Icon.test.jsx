import React from 'react';
import { render, screen } from '@testing-library/react';
import Icon from '../Icon';

describe('Icon component', () => {
  it('renders Icon component correctly', () => {
    render(<Icon />);

    expect(screen.container).toMatchSnapshot();
  });
});
