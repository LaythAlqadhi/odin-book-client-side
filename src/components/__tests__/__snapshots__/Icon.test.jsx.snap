// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon component > renders Icon component correctly 1`] = `undefined`;
